plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/expression/ImplementMinMax.java

```diff
@@ -55,6 +64,11 @@
         JdbcColumnHandle columnHandle = (JdbcColumnHandle) context.getAssignment(input.getName());
         verify(columnHandle.getColumnType().equals(aggregateFunction.getOutputType()));
 
+        // Remote database is case insensitive or sorts values differently from Trino
+        if (!isRemoteCaseSensitive && (columnHandle.getColumnType() instanceof CharType || columnHandle.getColumnType() instanceof VarcharType)) {
+            return Optional.empty();
```

Comment:
That's based on a wrong type.

For example, PostgreSQL's money and enum types are mapped to Trino varchar, while both will have different sorting properties.

Reply:
Since the rewrite rule is generic I think the only solution is to allow connectors to pass a list of `jdbcTypeName` through the constructor for types which should not be pushed down?

Or maybe add a static function to JdbcClient called `isCollationSensitive(JdbcColumnHandle)` and let connectors define their own impls? This method already exists in the PostgreSQL client btw and may be useful for others where we can pass explicit collations (MySQL once we drop 5.x).

Reply:
Yeah, i think we already use "is mapped to char or varchar" as a way to determine if it's potentially case insensitive (or collation-sensitive).
it may catch too much, but shouldn't catch too little, so it's fine.

live as is